src/scikit_build_core/_compat/importlib/resources.py

```diff
@@ -7,7 +7,17 @@
 else:
     from importlib.resources import files
 
-__all__ = ["files"]
+if sys.version_info < (3, 10):
+    # Readers were introduced in 3.10
+    from importlib_resources import readers
+elif sys.version_info < (3, 11):
+    # In 3.10 readers were implemented in importlib.readers
+    from importlib import readers
+else:
+    # From 3.11 it is accessed as importlib.resources.readers
+    from importlib.resources import readers
```

Comment:
This part is rather tricky since this can be expanded into 3 possibilities:
- `importlib_resources.MultiplexedPath`
- `importlib.readers.MultiplexedPath`
- `importlib.resources.readers.MultiplexedPath`

The latter 2 should work fine as-is, but the former one is tricky because in python 3.9 `readers` doesn't exist and it shouldn't be creating a `MultiplexedPath` when going through the `resource_reader`.  So it shouldn't merit adding `importlib_resources` as a dependency to python<3.10.

Maybe I should make a different compatibility to `importlib/readers.py` instead and make `MultiplexedPath` a dummy class?